### ISSUE TITLE
Component icons

### ISSUE THREAD

**Issue by perillosantana** [vtex-apps/store-header#69]
## Expected behavior
Nowadays we do not have the option of inserting icons based on client layouts, the idea is to have the option of inserting our source icons and manipulating the component to work with our icons.

## Current Behavior
the current behavior allows calling a font, but does not allow manipulating the component to insert our icon (font or img / svg), we are manipulating the element with before and after.

## Context
The idea is to achieve greater precision in relation to customer layouts

**Comment by matheusps** [comment]
Hey! You could change all the icons used on `header` or any other component by changing it's `svg` inside [`store-icons`](https://github.com/vtex-apps/store-icons). Its that what you're looking for ?

**Comment by perillosantana** [comment]
Hi @matheusps, with the "store-icons" can I customize an icon with any svg I want?

**Comment by matheusps** [comment]
Yep! Actually you can add an entirelly new `iconpack`. This is how you do it: 

1. Go to your [`store-theme/styles`](https://github.com/vtex-apps/store-theme/tree/master/styles)
2. Add a new folder called `iconpacks`
3. Create a new file `iconpack.svg` with your id's. [Like here](https://github.com/vtex-apps/store-icons/tree/master/styles/iconpacks).

Obs: You should not change the ID's, the store-icons use them on it's components.

**Comment by jpgutti** [comment]
@matheusps Are you saying that is up to us to build our own icon pack and make a pull request or we can do that in our project folder without the need of a PR?

**Comment by matheusps** [comment]
@jpgutti , You guys use the store-theme to build your own theme, right? The `iconpack` behaves the same. And yes, if you want different icons, is up to you to build a new iconpack that follows the same structure of the one inside `store-icons`.

**Comment by jpgutti** [comment]
@matheusps Awesome, thanks!

**Comment by matheusps** [comment]
You're welcome @perillosantana

**Comment by perillosantana** [comment]
@matheusps As I use a new one, I saw that inside the react folder there are the files calling the Icon, so when I use the iconpack I get the id of a guy that already exists, how do I create one from zero picking up from inside my iconpack?

[https://github.com/vtex-apps/store-icons/blob/master/react/IconSearch.tsx](url)

**Comment by micas06gua** [comment]
this does not work to exchange icons of the native components?

**Comment by micas06gua** [comment]
this does not work to exchange icons of the native components?

**Comment by matheusps** [comment]
@perillosantana, @micas06gua,

Take `IconSearch` as example: 
```
const IconSearch = (props: IconProps) => {
  return <Icon id="hpa-search" {...props} />
}
```

The Id `hpa-search` is defined on `iconpack.svg`, right ? (that why you may not change the ID) So, in your `store-theme` you do like,

> Yep @perillosantana . Actually you can add an entirelly new `iconpack`. This is how you do it:
> 
> 1. Go to your [`store-theme/styles`](https://github.com/vtex-apps/store-theme/tree/master/styles)
> 2. Add a new folder called `iconpacks`
> 3. Create a new file `iconpack.svg` with your id's. [Like here](https://github.com/vtex-apps/store-icons/tree/master/styles/iconpacks).
> 
> Obs: You should not change the ID's, the store-icons use them on it's components.

And YES, your new iconpack will be used instead of the original one contained on `store-icons`. Got it ?

**Comment by gustavo-vasconcellos** [comment]
@matheusps This component `Icon` is imported of which module?

**Comment by perillosantana** [comment]
@matheusps I modified the value of the id hpa-hamburguer-menu, but it does not reflect in the air as you can see in the images, can you validate if I'm doing something wrong?

Another question is, can I create an id and call with Icon?

[https://ibb.co/j5D5cZL](https://ibb.co/j5D5cZL)
[https://ibb.co/jzn4WnT](https://ibb.co/jzn4WnT)
[https://ibb.co/RPM0hz9](https://ibb.co/RPM0hz9)

**Comment by perillosantana** [comment]
@matheusps To test replacing the **hpa-hamburger-menu** path for the **hpa-search** id path, in the component I call the `import {IconMenu} from 'vtex.store-icons'` and the `<IconMenu size = {10} />`, but it continues to render the **hpa-hamburger-menu** icon, my folder respects the requested structure `/styles/iconpacks/iconpack.svg`, any idea why it does not overwrite?

Another question is, can I create an id and call with Icon?
